packages/address/src/address.ts

```diff
@@ -142,17 +142,23 @@ export default class Address extends AbstractAddress {
    * thrown if the input string is not nilsy and cannot be resolved to a valid address format
    * @returns a new `Address` instance
    */
-  static fromDynamicInput(addressId: string): Address {
-    if (isPublicKey(addressId)) {
-      return Address.fromPublicKey(addressId);
+  static fromDynamicInput(address: string | AbstractAddress): Address {
```

Comment:
I'll repeat myself in this regard, but I'd prefer if we could keep it as before, not overloading the `address` parameter type and keeping it as `string` only.

I know there are many places with this kind of overloading, and it might be tricky to adjust them all, but we should at least make an extra effort not to **_add new ones_**.

```suggestion
  static fromDynamicInput(addressId: string): Address {
```

Reply:
This will be removed on the PR to remove the object Address.

Reply:
@luizstacio Adding something only to be removed later isn't optimal.

@camsjams Don't the `Address.fromDynamicInput` method already know things for certain?

Reply:
If I remove it from here I would need to implement the same logic in the other place. This is why for now it makes sense to receive both.